lib/service/Service.ts

```diff
@@ -354,35 +355,45 @@ class Service extends EventEmitter {
           break;
       }
 
-      // resp, err := cmdLnd.SendPaymentSync(lncctx,&lnrpc.SendRequest{
-      //   DestString:deal.makerPubKey,
-      //   Amt:deal.makerAmount,
-      //   PaymentHash:deal.hash[:],
-      // })
-      // if err != nil{
-      //   err = fmt.Errorf("Got error sending  %d %v by taker - %v",
-      //         deal.makerAmount,deal.makerCoin.String(),err)
-      //   log.Printf(err.Error())
-      //   return nil, err
-      // }
-      // if resp.PaymentError != ""{
-      //   err = fmt.Errorf("Got PaymentError sending %d %v by taker - %v",
-      //       deal.makerAmount,deal.makerCoin.String(), resp.PaymentError)
-      //   log.Printf(err.Error())
-      //   return nil, err
-      // }
-
-      // this.logger.info('sendPayment response from maker to taker:' + JSON.stringify(resp))
-
-      return 'preImage to here';
-    }
+      if (!deal.makerPubKey) {
+        return 'makerPubKey is missing';
+      }
 
-	// If we are here we are the maker
-    this.logger.debug('Executing maker code');
+      const request = new lndrpc.SendRequest();
+      request.setAmt(deal.makerAmount);
+      request.setDestString(deal.makerPubKey);
+      request.setPaymentHashString(String(deal.r_hash));
+
+      try {
+        const response = await cmdLnd.sendPaymentSync(request);
+        if (response.getPaymentError()) {
+          this.logger.error('Got error from sendPaymentSync: ' + response.getPaymentError() + ' ' + JSON.stringify(request.toObject()));
+          return response.getPaymentError();
+        }
+
+        if (!response.getPaymentPreimage()) {
+        // TODO: Handle this unexpected case, if we are the maker we should have the preimage
+          return 'did not get the preImage';
+        }
+        const hexString = Buffer.from(response.getPaymentPreimage_asB64(), 'base64').toString('hex');
```

Comment:
Is there any particular reason you're converting this to hex rather than leaving it in Base64?

Reply:
The caller, LND resolver, expects a string and not a base64 encoded string.

Reply:
Gotcha.